### PR TITLE
Fixes for hashimoto algo""

### PR DESCRIPTION
**Pull request by rsdevgun16e**
My change was pushed to main repo accidently. This is the PR for same change reverted in main.

### CODE REVIEW

**Comment by ediston** [comment]
Please add comments
Why "Revert "Revert "Fixes for hashimoto algo ?"

File: src/crypto/egihash.cpp [hunk 1]
```diff
@@ -361,6 +362,19 @@ namespace egihash
 	::std::string get_seedhash(uint64_t const block_number)
 	{
 		::std::string s(epoch0_seedhash, size_epoch0_seedhash);
+//		// convert seedhash to hex
```
Comment (ediston) [inline src/crypto/egihash.cpp:365]: why are we doing this ?
will we come back to this?

File: src/crypto/egihash.cpp [hunk 2]
```diff
@@ -941,51 +955,63 @@ namespace egihash
...
-			for (size_t i = 0; i < mix.size(); i += 4)
+			for (uint32_t i = 0; i < mix.size(); i += 4)
 			{
 				cmix.push_back(node(fnv(fnv(fnv(mix[i].hword, mix[i+1].hword), mix[i+2].hword), mix[i+3].hword)));
 			}
 
-			auto v = sha3_256(s);
+			auto combined = s;
+			combined.insert(combined.end(), cmix.begin(), cmix.end());
+			auto v = sha3_256(combined);
```
Comment (ediston) [inline src/crypto/egihash.cpp:998]: why are we creating a new vector ?

File: src/crypto/egihash.cpp [hunk 3]
```diff
@@ -941,51 +955,63 @@ namespace egihash
 #endif // 0
 
 	// TODO: unify light & full implementation
-	namespace full
+	namespace hashimoto
```
Comment (ediston) [inline src/crypto/egihash.cpp:958]: good job + 1

File: src/crypto/egihash.cpp [hunk 4]
```diff
@@ -997,45 +1023,9 @@ namespace egihash
 	{
 		result_t hash(cache_t const & cache, void const * input_data, cache_t::size_type input_size)
 		{
-			auto const n = dag_t::get_full_size((cache.epoch() * constants::EPOCH_LENGTH) + 1) / constants::HASH_BYTES;
```
Comment (ediston) [inline src/crypto/egihash.cpp:1000]: how's it still a light hash ? if we are calling full internally 
:) 
the more the info in the commit message.. better it is

File: src/crypto/keccak-tiny.c [hunk 5]
```diff
@@ -157,7 +157,7 @@ static inline int hash(uint8_t* out, size_t outlen,
     if (outlen > (bits/8)) {                                      \
       return -1;                                                  \
     }                                                             \
-    return hash(out, outlen, in, inlen, 200 - (bits / 4), 0x06);  \
+    return hash(out, outlen, in, inlen, 200 - (bits / 4), 0x01);  \
```
Comment (ediston) [inline src/crypto/keccak-tiny.c:160]: why 0x01 ?

File: src/crypto/egihash.h [hunk 6]
```diff
@@ -157,8 +160,7 @@ namespace egihash
 	*	The epoch0_seedhash should be set to a randomized set of 32 bytes for a given crypto currency.
 	*	This represents a keccak-256 hash that will be used as input for building the DAG/cache.
 	*/
-	// TODO: randomized seedhash not zero seedhash
-	static constexpr char epoch0_seedhash[] = "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00";
+	static constexpr char epoch0_seedhash[] = "\xa8\x49\x4b\xb2\x89\x5b\xd7\xed\x18\xbb\x39\xb7\xb2\x8a\xf5\x1d\xec\x51\xf7\xca\xd3\x30\xc1\x68\xf1\xbd\x1c\x90\xe7\x61\x4c\x32";
```
Comment (ediston) [inline src/crypto/egihash.h:163]: random ?

**Comment by ediston** [comment]
Please also have a look at the commit message 
---> Fixes for hashimoto algo""

File: src/crypto/egihash.cpp (hunk 4 above)
Comment (rsdevgun16e) [reply src/crypto/egihash.cpp:1000]: This is just to get size and below function calc dag item.

File: src/crypto/egihash.cpp (hunk 2 above)
Comment (rsdevgun16e) [reply src/crypto/egihash.cpp:998]: this was during test, anyway I fixed but this is nothing there is lot to be done for performance!

File: src/crypto/egihash.cpp (hunk 1 above)
Comment (rsdevgun16e) [reply src/crypto/egihash.cpp:365]: Ryan's idea is to have filename readable if we do this. After discussion we dont need to. I will change the to create readable file name along with compatible with etihash algo implementation

File: src/crypto/keccak-tiny.c (hunk 5 above)
Comment (rsdevgun16e) [reply src/crypto/keccak-tiny.c:160]: must be 1, this branch is not upto date looks like.

File: src/crypto/egihash.h (hunk 6 above)
Comment (rsdevgun16e) [reply src/crypto/egihash.h:163]: Ryan chose to have this initial seed hash. I guess after consensus

**Pull request closed without merging**